### DISCUSSION TITLE
Quick Code Folding File Overview

### DISCUSSION THREAD

**Discussion by sebastiansandqvist** [zed-industries/zed#36949] [Feature Requests]
### Description

An awesome feature in the [makepad.dev](https://makepad.dev/) editor is the ability to fold all but the two outermost blocks of code in a file (using the <kbd>alt</kbd> key), zooming the rest of the code out. I believe that since Zed has deep LSP integrations and runs its rendering through the GPU, it should be well suited to implement something like this.

https://github.com/user-attachments/assets/d7c0f34f-c6b6-43fb-8dbc-a4bfb4de494a

The first time I saw this feature I was amazed by how useful it is for getting a high level overview of a file at a glance and for quickly zooming around a file. The <kbd>cmd-shift-O</kbd> outline view in Zed is similar, but it's just out of the way enough that I don't find myself reaching for it nearly as often. I think this is one of those features that you have to play around with a bit to fully appreciate. I see it as a more ergonomic alternative to Zed's outline view. An interesting detail is that you can place the cursor anywhere within the zoomed out parts of the code and the editor still responds as though the code were unfolded.

Any interest in adding such a feature?

**Comment by SomeoneToIgnore** [comment]
Zed also has an outline panel now, so that might be some sort of alternative (and it's not a modal, so can be always opened on a side)

**Comment by notpeter** [comment]
Related: 
- #5142

**Comment by notpeter** [comment]
Interesting.  Basically this would a mouse/keyboard action which folded at various levels (like the  `["editor::FoldAtLevel", 2]` actions) and replace them with a minimap style place holder. 

This is not something we intend to prioritize in the nearterm, but thanks for filing.
